In file included from dynamic-link.h:21,
                 from dl-reloc.c:153:
../sysdeps/powerpc/dl-machine.h: In function '_dl_relocate_object':
../sysdeps/powerpc/dl-machine.h:399: error: invalid storage class for function 'elf_machine_rela_relative'
make[2]: *** [/home/dank/queue/jobdir.dual2/crosstool-dev/build/powerpc-750-linux-gnu/gcc-4.0-20050305-glibc-2.2.5/build-glibc/elf/dl-reloc.o] Error 1
--- glibc-2.2.5/sysdeps/powerpc/dl-machine.h.old	Thu Mar 17 13:28:17 2005
+++ glibc-2.2.5/sysdeps/powerpc/dl-machine.h	Thu Mar 17 13:29:03 2005
@@ -393,7 +393,14 @@
 			    reloc_addr, finaladdr, rinfo);
 }
 
-static inline void
+#if __GNUC__ >= 4
+  auto inline void
+#else
+  static inline void
+#endif
+#if __GNUC__ >= 4 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 2)
+  __attribute ((always_inline))
+#endif
 elf_machine_rela_relative (Elf32_Addr l_addr, const Elf32_Rela *reloc,
 			   Elf32_Addr *const reloc_addr)
 {
